### DISCUSSION TITLE
With buildx create, how do you check if a builder already exists?

### DISCUSSION THREAD

**Discussion by NiklasBr** [docker/buildx#1319] [Q&A] [answered]
Before running something like: this:

```bash
docker buildx create --name $MY_BUILD_NAME --platform $MY_BUILD_PLATFORMS --driver docker-container --use
```

I would like to check whether or not a builder with that name already exists or not. Is there a way to do that?

**Comment by jedevc** [comment, marked as answer]
You can use the `docker buildx inspect` command:

e.g.

```bash
$ docker buildx inspect default
Name:   default
Driver: docker

Nodes:
Name:      default
Endpoint:  desktop-linux
Status:    running
Buildkit:  20.10.17
Platforms: linux/amd64, linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/arm/v7, linux/arm/v6

$ buildx inspect non-existent
ERROR: no builder "non-existent" found
```

If the builder cannot be found, then the return value of the command will a non-zero exit code.

**Comment by treefern** [comment]
A quicker option seems to be running `docker buildx ls` and grepping for the expected container or driver type.
e.g.
```
if [ "$(docker buildx ls | grep docker-container)" -eq 1 ]; then
    docker buildx create --name buildx-container --driver=docker-container
fi
 ```

**Comment by sebastienmascha** [comment]
``For a script as @treefern  suggested, we would want to use an if condition.

I believe that I have improved what @treefern suggested:
- Avoids string-to-number conversion issues.
- Exit Status Handling: The exit status of the grep command directly indicates whether the search term was found. By checking this status, we can simplify the logic and avoid potential pitfalls with string handling.

The improved script uses the exit status of the grep command, which is a more reliable and straightforward approach. Here's how it works:
``` sh
if docker buildx ls | grep -q sr-builder; then
  echo "Builder exists"
else
  echo "Builder does not exist"
fi
```

- `docker buildx ls | grep -q sr-builder`:
  - `docker buildx ls` lists the builders.
  - `grep -q sr-builder` searches for the string `sr-builder` quietly (without printing to the output).
  - The exit status of `grep` is `0` if the string is found and `1` if it is not found.

**Comment by jamesalucas** [comment]
Could you use the output status to short circuit the create command?

```
docker buildx inspect container || docker buildx create --name container --driver=docker-container
```

**Reply by NiklasBr** [reply]
That looks like a nice one-liner!

```shell
docker buildx inspect desktop-linux2 || docker buildx create --name desktop-linux2 --driver=docker-container
ERROR: no builder "desktop-linux2" found
desktop-linux2
```
```shell
docker buildx inspect desktop-linux2                                                                                   
Name:          desktop-linux2
Driver:        docker-container
Last Activity: 2024-10-17 13:41:37 +0000 UTC

Nodes:
Name:                  desktop-linux20
Endpoint:              desktop-linux
Status:                inactive
BuildKit daemon flags: --allow-insecure-entitlement=network.host
```

**Reply by TomK** [reply]
you can silence the output like this
```sh
(docker buildx inspect builder || docker buildx create --name builder --use) >/dev/null 2>&1
```
